libp2p/switch.nim

```diff
@@ -250,6 +231,33 @@ proc stop*(s: Switch) {.async.} =
 
   trace "Switch stopped"
 
+proc start*(s: Switch): Future[seq[Future[void]]] {.async, gcsafe.} =
+  trace "starting switch for peer", peerInfo = s.peerInfo
+  var startFuts: seq[Future[void]]
+  for t in s.transports: # for each transport
+    for i, a in s.peerInfo.addrs:
+      if t.handles(a): # check if it handles the multiaddr
+        let transpStart = t.start(a)
+        startFuts.add(transpStart)
+        try:
+          await transpStart
+          s.peerInfo.addrs[i] = t.ma # update peer's address
```

Comment:
if there are two transports, won't this overwrite `addrs[i]`?

Reply:
The idea is that each transport handle a different set of addresses, but it's not handled correctly ATM, we're working on it in #598